std-bits/table/src/main/java/org/enso/table/data/column/storage/DateTimeStorage.java

```diff
@@ -69,4 +76,37 @@ public StorageType getType() {
   public Builder createDefaultBuilderOfSameType(int capacity) {
     return new DateTimeBuilder(capacity);
   }
+
+  @Override
+  public Storage<?> cast(StorageType targetType, CastProblemBuilder castProblemBuilder) {
+    if (targetType instanceof DateType) {
```

Comment:
should we be reporting this as "lossy" if not midnight?

Reply:
Hmmm. This is the general question of where we place the border between reporting Lossy and not.

I thought it is more of a case like `float->int` or `varchar->char(10)` - we don't report lossy there because we know there is some data loss (we report only if it is unexpected - like `float->int` and `float` is exceeding the range; but just losing the fractional part is not 'considered lossy' in this model).

So I thought that losing the time-of-day part here is more like losing the fractional part - expected.

I'm not sure really, it feels like both choices have some + and -.

Reply:
Will rename `Lossy_Conversion` to `Conversion_Failure` as discussed.